packages/pigeon/lib/generator_tools.dart

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+import 'ast.dart';
+
+/// Read all the content from [stdin] to a String.
+String readStdin() {
+  final List<int> bytes = <int>[];
+  int byte = stdin.readByteSync();
+  while (byte >= 0) {
+    bytes.add(byte);
+    byte = stdin.readByteSync();
+  }
+  return utf8.decode(bytes);
+}
+
+/// A helper class for managing indentation, wrapping a [StringSink].
+class Indent {
```

Comment:
Ah. Templating it would be kind of hard at this point. It's fine to leave it as is.

Reply:
ack